grafana/resource_dashboard.go

```diff
@@ -211,12 +213,17 @@ func ReadDashboard(ctx context.Context, d *schema.ResourceData, meta interface{}
 		return diag.FromErr(err)
 	}
 
+	configJSON := d.Get("config_json").(string)
+
+	// Check if the SHA256 of configJSON is stored in the state
+	isSHA256, _ := regexp.MatchString(`[A-Fa-f0-9]{64}`, configJSON)
```

Comment:
I'm not keen on compiling the regular expression every time this is evaluated. Can you cache this at the package level and then call the `MatchString` function on that?

Also, while at it, can we change the regular expression to have start and end anchors, as in `^[A-Fa-f0-9]{64}$`?

I've made a quick benchmark and the performance, both in memory and speed, is too much to ignore:

```
$ gotest -run=XXX -benchmem -bench='^BenchmarkSHA$' -benchtime=1000000x
goos: darwin
goarch: amd64
pkg: github.com/inkel/misc-go-benchmarks
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkSHA/sha256/uncached-12                  1000000              4495 ns/op            6999 B/op         23 allocs/op
BenchmarkSHA/sha256/cached-12                    1000000               603.8 ns/op             0 B/op          0 allocs/op
BenchmarkSHA/sha256/cached+anchors-12            1000000               613.0 ns/op             0 B/op          0 allocs/op
BenchmarkSHA/json64/uncached-12                  1000000             13037 ns/op            6985 B/op         23 allocs/op
BenchmarkSHA/json64/cached-12                    1000000              9241 ns/op               0 B/op          0 allocs/op
BenchmarkSHA/json64/cached+anchors-12            1000000                40.24 ns/op            0 B/op          0 allocs/op
BenchmarkSHA/json/uncached-12                    1000000              7295 ns/op            7000 B/op         23 allocs/op
BenchmarkSHA/json/cached-12                      1000000              3172 ns/op               0 B/op          0 allocs/op
BenchmarkSHA/json/cached+anchors-12              1000000                39.25 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/inkel/misc-go-benchmarks     39.038s
```

Here, `json64` refers to a string with a length of 64 characters, same as a SHA256 hash. Note particularly the case of passing a string that is not a SHA256 hash while the regular expression is cached and includes anchors.